lib/aggregate/bitfield.rb

```diff
@@ -0,0 +1,81 @@
+# A string representation of a fixed length array of nillable booleans.
+# A length limited version of the class is created by calling the limit method.
+module Aggregate
+  class Bitfield
+    include Comparable
+
+    def initialize(string_form)
+      @string = string_form
+    end
+
+    def [](index)
+      check_index_limit(index)
+      to_boolean(@string[index])
+    end
+
+    def []=(index, value)
+      check_index_limit(index)
+      if index > @string.size
+        @string = @string.ljust(index)
```

Comment:
I would probably just skip the if since ljust(index) doesn't do anything when `index <= @string.size`

So just 
```
check_index_limit(index)
@string = @string.ljust(index)
@string[index] = to_character(value)
```
..